workers/unity/Packages/io.improbable.gdk.tools/GdkToolsConfigurationWindow.cs

```diff
@@ -224,5 +248,22 @@ private void DrawCustomSnapshotDir()
                 }
             }
         }
+
+        private void Update()
+        {
+            TrySaveChanges();
+        }
+
+        private void TrySaveChanges()
+        {
+            var timeSinceLastSave = DateTime.Now - lastSaveTime;
+            if (!hasUnsavedData || timeSinceLastSave < FileSavingInterval)
+            {
+                return;
+            }
+
+            toolsConfig.Save();
+            lastSaveTime = DateTime.Now;
+        }
```

Comment:
I think there's a race here where we could save invalid data.

1. User inputs some valid value, `hasUnsavedData` is set to true.
2. User changes a value to invalid. 
3. The interval passes and we are able to save.
4. Invalid data is saved

There should be a check against any config errors before we are able to save: 

```csharp
if (!hasUnsavedData || timeSinceLastSave < FileSavingInterval || configErrors.Any())
```

Reply:
Sounds good, making this change.